spec/factories/report.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :v2_report, class: 'V2::Report' do
+    to_create do |instance, context|
+      policy = FactoryBot.create(
+        :v2_policy,
+        :for_tailoring,
+        title: context.title,
+        os_major_version: context.os_major_version,
+        compliance_threshold: context.compliance_threshold,
+        business_objective: context.business_objective,
+        account: context.account,
+        supports_minors: context.supports_minors
+      )
+
+      policy.os_minor_versions.each do |version|
+        FactoryBot.create(
+          :v2_test_result,
+          account: context.account,
+          tailoring: FactoryBot.create(
+            :v2_tailoring,
+            policy: policy,
+            os_minor_version: version
+          )
+        )
+      end
+
+      instance.id = policy.id
+      instance.attributes = V2::Policy.find(instance.id).attributes
+      instance.reload
+    end
+
+    transient do
+      os_major_version { 7 }
+      business_objective { 'safety' }
+      compliance_threshold { 90 }
+      title { 'aaa' }
+      supports_minors { [0] }
+    end
```

Comment:
I think there are additional params necessary:
* how many systems are in the report
* how many of these systems have results
* how many of these systems are supported
* how many of these systems have compliant results

I think you should come up with a consistent set of params for this...

Reply:
The key here is on the consistency, the relationship between these counts is not linear and might need a hierarchical parametrization.